projects/slap_manipulation/src/slap_manipulation/policy/voxel_grid.py

```diff
@@ -0,0 +1,222 @@
+from functools import reduce
+from operator import mul
+
+import torch
+from torch import nn
+
+MIN_DENOMINATOR = 1e-12
+INCLUDE_PER_VOXEL_COORD = False
+
+
+class VoxelGrid(nn.Module):
```

Comment:
Everything in this class needs to be documented with docstrings and type hints - very hard to figure out how I would use this right now

Reply:
I can try to annotate it a bit, this is taken from PerAct as is. Added their info.